tests/tracer/test_logging.py

```diff
@@ -0,0 +1,518 @@
+import os
+import re
+
+import pytest
+
+from ddtrace.internal.compat import PY2
+
+
+LOG_PATTERN = r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} \w{1,} \[\S{1,}\] \[\w{1,}.\w{2}:\d{1,}\] - .{1,}$"
+
+
+@pytest.mark.parametrize("dd_trace_debug", ["true", "false", None])
+@pytest.mark.parametrize("dd_trace_log_file_level", ["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG", None])
+@pytest.mark.parametrize("dd_trace_log_file", ["example.log", None])
+def test_unrelated_logger_loaded_first_in_debug(
+    dd_trace_debug, dd_trace_log_file_level, dd_trace_log_file, run_python_code_in_subprocess, tmpdir
+):
+    """
+    When the tracer is imported after logging has been configured in debug mode,
+        the ddtrace logger does not override any custom logs settings.
+    """
+    env = os.environ.copy()
+    if dd_trace_debug is not None:
+        env["DD_TRACE_DEBUG"] = dd_trace_debug
+
+    if dd_trace_log_file_level is not None:
+        env["DD_TRACE_LOG_FILE_LEVEL"] = dd_trace_log_file_level
+
+    if dd_trace_log_file is not None:
+        env["DD_TRACE_LOG_FILE"] = tmpdir.strpath + dd_trace_log_file
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+import ddtrace
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+ddtrace_logger = logging.getLogger('ddtrace')
+ddtrace_logger.critical('ddtrace critical log')
+ddtrace_logger.warning('ddtrace warning log')
+"""
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+
+    if dd_trace_log_file is not None:
+        assert out == b""
+
+        if PY2 and dd_trace_debug == "true":
+            assert 'No handlers could be found for logger "ddtrace' in err
+        else:
+            assert err == b""
+
+        with open(tmpdir.strpath + dd_trace_log_file) as file:
+            first_line = file.readline()
+            assert len(first_line) > 0
+            assert re.search(LOG_PATTERN, first_line) is not None
+    else:
+        if PY2:
+            assert 'No handlers could be found for logger "ddtrace' in err
+        else:
+            assert b"ddtrace warning log" in err
+        assert out == b""
+
+
+@pytest.mark.parametrize("dd_trace_debug", ["true", "false", None])
+@pytest.mark.parametrize("dd_trace_log_file_level", ["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG", None])
```

Comment:
Do we need to test across all log levels here? If this test asserts that we do not override any custom log settings should be able to test with just the debug level.

Reply:
should we be only testing with `dd_trace_debug=True`? If not we should update the doc string to something like 
`When the tracer is imported before logging has been configured the ddtrace logger does not override any custom logs settings`. 

Also I removed the `dd_trace_debug` parameter from this test and all the test cases still passed. I don't think it's need in this test. If it is then the test might be broken.